firefox_ui_harness/runtests.py

```diff
@@ -29,9 +48,46 @@ def __init__(self, *args, **kwargs):
         self.test_handlers = [FirefoxTestCase]
 
 
-def run():
-    cli(runner_class=ReleaseTestRunner, parser_class=ReleaseTestParser)
+def startTestRunner(runner_class, options, tests):
+    try:
+        # If the specified binary is an installer it needs to be installed
+        installed = False
+        if mozinstall.is_installer(options.binary):
+            dest_folder = tempfile.mkdtemp()
+
+            options.logger.info('Installing build "%s" to "%s"...' % (options.binary,
+                                                                      dest_folder))
+            install_folder = mozinstall.install(options.binary, dest_folder)
+            installed = True
+            options.binary = mozinstall.get_binary(install_folder, 'firefox')
+
+        runner = runner_class(**vars(options))
+        runner.run_tests(tests)
+
+    finally:
+        # Ensure to uninstall the binary if it has been installed before
+        if installed:
+            options.logger.info('Uninstalling build at "%s"...' % install_folder)
+            mozinstall.uninstall(install_folder)
+
+    return runner
+
+
+def cli(runner_class=FirefoxUITestRunner, parser_class=FirefoxUIOptions):
```

Comment:
I see. I guess this is fine to do, we can set our own logging options and do our own command line stuff later here, too.